### ISSUE TITLE
with activated "XY Hole/Contour compensation" there are offsets

### ISSUE THREAD

**Issue by Lusbueb69** [bambulab/BambuStudio#4727]
### Bambu Studio Version

1.9.5.51

### Where is the application from?

Bambu Lab Official website

### OS version

Windows 11 22H2

### Additional system information

no performance issue

### Printer

Bamu Lab X1 Carbon

### How to reproduce

1. load Model
2. activate XY Hole/Contour compensation (0.093/0.109)
3. slice Model

### Actual results

there are offsets and the Model is not accurate, because the offsets (Yellow marked)
![Aufnahme1](https://github.com/user-attachments/assets/b01888ef-0ebf-4784-b9b8-8e015dea7a89)
![Aufnahme2](https://github.com/user-attachments/assets/9868fa48-fd77-42bc-a719-3550ba389173)

### Expected results

without XY Hole/Contour compensation, no offsets
![Aufnahme3](https://github.com/user-attachments/assets/d0669b9f-bd19-4bec-b691-90f833efc878)

### Project file & Debug log uploads

[logs.zip](https://github.com/user-attachments/files/16825346/logs.zip)
[Project.zip](https://github.com/user-attachments/files/16825347/Project.zip)

### Checklist of files to include

- [X] Log file
- [X] Project file

**Comment by QingZhangBambu** [comment]
@Lusbueb69  yes, it really a improve point. but it tricky to do that. In 2D， we don't know which is contour which is holes.
